### PR TITLE
Plastic Curtains

### PR DESCRIPTION
**Pull request by Konater**
Stolen from polaris, honk
You can construct/deconstruct them for four plastic sheets, I'm guessing plastic flaps won't go missing from now on
I look forward to the line comments
curtains

### CODE REVIEW

File: code/game/objects/structures/curtains.dm [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+	density = 0
+
+/obj/structure/curtain/open
+	icon_state = "open"
+	layer = SHOWER_CLOSED_LAYER
+	opacity = 0
+
+/obj/structure/curtain/bullet_act(obj/item/projectile/P, def_zone)
+	if(!P.nodamage)
+		visible_message("<span class='warning'>[P] tears [src] down!</span>")
```
Comment (9600bauds) [inline code/game/objects/structures/curtains.dm]: \the [src]

File: code/game/objects/structures/curtains.dm [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+		qdel(src)
+	else
+		..(P, def_zone)
```
Comment (PJB3005) [inline code/game/objects/structures/curtains.dm]: Arguments automatically get passed down, no need to be explicit.

File: code/game/objects/structures/curtains.dm [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+		playsound(src.loc, 'sound/items/Wirecutter.ogg', 50, 1)
+		if(do_after(user, src, 10))
+			to_chat(user, "<span class='notice'>You cut the shower curtains down.</span>")
+			var/obj/item/stack/sheet/mineral/plastic/A = getFromPool(/obj/item/stack/sheet/mineral/plastic, get_turf(src))
+			A.amount = 4
+			qdel(src)
+		return
+	else
+		src.attack_hand(user)
+	return
```
Comment (PJB3005) [inline code/game/objects/structures/curtains.dm]: Trailing return.

File: code/game/objects/structures/curtains.dm [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+	if(opacity)
+		icon_state = "closed"
+		layer = SHOWER_CLOSED_LAYER
+	else
+		icon_state = "open"
+		layer = SHOWER_OPEN_LAYER
+
+/obj/structure/curtain/attackby(obj/item/W, mob/user)
+	if(iswirecutter(W))
+		playsound(src.loc, 'sound/items/Wirecutter.ogg', 50, 1)
```
Comment (PJB3005) [inline code/game/objects/structures/curtains.dm]: Remove `src.`

File: code/game/objects/structures/curtains.dm [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+		if(do_after(user, src, 10))
+			to_chat(user, "<span class='notice'>You cut the shower curtains down.</span>")
+			var/obj/item/stack/sheet/mineral/plastic/A = getFromPool(/obj/item/stack/sheet/mineral/plastic, get_turf(src))
+			A.amount = 4
+			qdel(src)
+		return
+	else
```
Comment (PJB3005) [inline code/game/objects/structures/curtains.dm]: No need.

File: code/game/objects/structures/curtains.dm [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+
+/obj/structure/curtain/attackby(obj/item/W, mob/user)
+	if(iswirecutter(W))
+		playsound(src.loc, 'sound/items/Wirecutter.ogg', 50, 1)
+		if(do_after(user, src, 10))
+			to_chat(user, "<span class='notice'>You cut the shower curtains down.</span>")
+			var/obj/item/stack/sheet/mineral/plastic/A = getFromPool(/obj/item/stack/sheet/mineral/plastic, get_turf(src))
+			A.amount = 4
+			qdel(src)
+		return
```
Comment (ComicIronic) [inline code/game/objects/structures/curtains.dm]: Return 1 on success

File: code/game/objects/structures/bedsheet_bin.dm [hunk 7]
```diff
@@ -50,7 +50,7 @@ LINEN BINS
 /obj/item/weapon/bedsheet/attack_self(mob/user as mob)
 	user.drop_item(src, force_drop = 1)
 	if(layer == initial(layer))
-		layer = 5
+		layer = MOB_LAYER + 0.1
```
Comment (ComicIronic) [inline code/game/objects/structures/bedsheet_bin.dm:53]: NON
ATOMIC

**Comment by ComicIronic** [comment]
Looks good.

**Comment by Probe1** [comment]
time to hide behind curtains and jump out at motherfuckers

**Pull request merged**